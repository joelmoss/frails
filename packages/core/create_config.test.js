const serializer = require('jest-serializer-path')

const { createConfig, createConfigBlock } = require('./create_config')

describe('createConfig', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer(serializer)
  })

  test('without args', () => {
    expect(createConfig()).toMatchSnapshot()
  })

  test.only('with an object arg', () => {
    // Act
    const config = createConfig({ devtool: 'cheap-module-eval-source-map' })

    expect(config).toMatchSnapshot()
    expect(config.devtool).toBe('cheap-module-eval-source-map')
  })

  test('with block args', () => {
    // Setup
    const aBlock = createConfigBlock('aBlock', (options, config) => {
      config.aBlock = true
    })

    const anotherBlock = createConfigBlock('anotherBlock', ({ name }, config) => {
      config.anotherBlock = { name }
    })

    // Act
    const config = createConfig(aBlock, anotherBlock({ name: 'Joel' }))

    // Expect
    expect(config.aBlock).toBe(true)
    expect(config.anotherBlock).toEqual({ name: 'Joel' })
  })

  test('included plugins', () => {
    // Act
    const config = createConfig()
    const plugins = config.plugins.map(plugin => plugin.constructor.name)

    // Expect
    expect(plugins).toEqual(['WebpackFixStyleOnlyEntriesPlugin', 'WebpackAssetsManifest'])
  })
})
