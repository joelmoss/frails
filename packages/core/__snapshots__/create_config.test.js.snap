// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createConfig with an object arg 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "devtool": "cheap-module-eval-source-map",
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;

exports[`createConfig without args 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;
